Kernel/Heap/kmalloc.cpp

```diff
@@ -204,6 +227,22 @@ struct KmallocGlobalData {
                 return ptr;
         }
 
+        // NOTE: This size calculation is a mirror of kmalloc_aligned(KmallocSlabBlock)
+        if (size <= KmallocSlabBlock::block_size * 2 + sizeof(ptrdiff_t) + sizeof(size_t)) {
+            // FIXME: We should propagate a freed pointer, to find the specific subheap it belonged to
+            //        This would save us iterating over them in the next step and remove a recursion
+            bool did_purge = false;
+            for (auto& slabheap : slabheaps) {
+                if (slabheap.try_purge()) {
+                    dbgln_if(KMALLOC_DEBUG, "Kmalloc purged block(s) from slabheap of size {} to avoid expansion", slabheap.slab_size());
```

Comment:
Can this not be a standard dbgln, like heap expansion, or is this very spammy?

Reply:
Heap expansion is also a dbgln_if, that's why I chose this here, too